Software/lib/ATECC608A-TNGLORA/atecc608a-tnglora-se-hal.c

```diff
@@ -25,16 +25,35 @@
 
 #include "atca_hal.h"
 #include "atca_device.h"
+#include "atca_basic.h"
 #include "atca_status.h"
 #include "stm32wlxx_hal_dma.h"
 #include "stm32wlxx_hal_i2c.h"
 #include "GNSE_bsp.h"
 
-uint32_t ATECC608ASeHalGetRandomNumber( void )
+uint32_t ATECC608ASeHalGetRandomNumber(void)
 {
-    // Not Implemeted
-    //TODO: Add suppport for random generation, see https://github.com/TheThingsIndustries/generic-node-se/issues/137
-    return 0;
+    uint8_t rand_out[RANDOM_NUM_SIZE];
+    uint32_t rand_ret;
+    /* Check if the load switch is powering the SE before controlling it */
+    int32_t ls_state = GNSE_BSP_LS_GetState(LOAD_SWITCH_SENSORS);
+    if (ls_state == GPIO_PIN_RESET)
+    {
+        GNSE_BSP_LS_On(LOAD_SWITCH_SENSORS);
+        HAL_Delay(LOAD_SWITCH_SENSORS_DELAY_MS);
+    }
```

Comment:
This section should not part of SE HAL
The device driver implementation assumes a powered device and this is the function of the LPM to synchronize sensor activation with secure element requests.